tests/test-addons.py

```diff
@@ -359,3 +360,27 @@ def test_rook_ceph_addon(self):
         validate_rook_ceph()
         print("Disabling Rook Ceph")
         microk8s_disable("rook-ceph")
+
+    def test_rook_ceph_integration(self):
+        """
+        Test Rook Ceph integration.
+        """
+        try:
+            subprocess.check_call("grep -E lxc /proc/1/environ /proc/cpuinfo".split())
```

Comment:
This could also fail for reasons like missing permissions etc.
Maybe `subprocess.check_output` would be better here?
e.g. something like this
```python
try:
    check_output("grep -E lxc /proc/1/environ /proc/cpuinfo".split())
except CalledProcessError as exc:
    # output will be empty if grep yields no result
    if exc.output:
        raise exc
```

Reply:
That is a fair point. Do we really want to differentiate between errors here? Have a look at this:
```
>>> try:
...     subprocess.check_output("grep -E lxc /proc/1/environ /proc/cpuinfo".split())
...     print("MicroCeph integration test cannot run on LXC. Skipped.")
... except subprocess.CalledProcessError as exc:
...     # output will be empty if grep yields no result
...     print("Failure")
...     print(exc)
...     print(exc.stdout)
...     print(exc.stderr)
...     print(exc.returncode)
...     if exc.stdout or exc.stderr:
...         print("Raise")
...         raise exc
... 
grep: /proc/1/environ: Permission denied
Failure
Command '['grep', '-E', 'lxc', '/proc/1/environ', '/proc/cpuinfo']' returned non-zero exit status 2.
b''
None
2
>>> try:
...     subprocess.check_output("grep -E swdsadas /home/jackal/Notes.txt".split())
...     print("MicroCeph integration test cannot run on LXC. Skipped.")
... except subprocess.CalledProcessError as exc:
...     # output will be empty if grep yields no result
...     print("Failure")
...     print(exc)
...     print(exc.stdout)
...     print(exc.stderr)
...     print(exc.returncode)
...     if exc.stdout or exc.stderr:
...         print("Raise")
...         raise exc
... 
Failure
Command '['grep', '-E', 'swdsadas', '/home/jackal/afile.txt']' returned non-zero exit status 1.
b''
None
```
In both cases where the grep fails to find any lines or when it does not have enough permissions the exc.output is the same empty string.

I suggest we keep it simple, here. If we do not detect we run on lxc for any reason lets proceed and let the test fail if there is any other error.

Reply:
Fair enough!